docs/django/image_field.md

```diff
@@ -0,0 +1,147 @@
+        fields = [     
+        #... , 
+        "cover_image", 
+        #... ,
+        ]
+````
+
+## API View
+
+````python
+class BookInformationViewset(viewsets.ModelViewSet):
+    queryset = BookInformation.objects.all()
+    serializer_class = BookInformationCreateUpdateSerializer
+````
+
+## ImageField Views Tests
+
+To test the views that involve the `cover_image` attribute you can simply generate an image based on the
+implemented factory for the cover image just like the example below.
+
+````python
+from PIL import Image
+import shutil
+from ..fixtures import REMOVE_MEDIA_FILES_PATH
+
+class BookCreateViewTestCase(APITestCase):
+    def setUp(self):
+        #...
+        self.data = {
+            "cover_image": #...
+            #...
+        }
+
+    def tearDown(self):
+        shutil.rmtree(REMOVE_MEDIA_FILES_PATH)
+
+    def test_create_with_image(self):
+        cover = BookInformationFactory.create().cover_image
+
+        self.data["cover_image"] = cover
+        res = self.client.post(self.url, self.data, format="multipart")
+
+        self.assertEqual(res.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(
+            BookInformation.objects.first().cover_image.read(), BookInformation.objects.last().cover_image.read()
+        )
+
+    def test_create_with_image_failure(self):
+        self.data["cover_image"] = Image.new(mode="RGB", size=(20, 20))
+
+        res = self.client.post(self.url, self.data, format="multipart")
+
+        self.assertEqual(res.status_code, status.HTTP_400_BAD_REQUEST)
+
+```` 
+
+It is crucial that the `format` option in set to `"multipart"` for the test to work. 
+It is also important to have the `tearDown` method implemented in order to delete the images
+saved on the folder refered previously (`.../library_project/media/covers/`).
+
+## ImageField Views Tests using temp files
+
+Other way of implementing a view test that includes an ImageField is with the `SimpleUploadedFile()` method.
+
+The code below will generate an Image with the PIL Library, save it in a temporary file,
+process it with `SimpleUploadedFile()` method and then make the POST request.
+
+In this case the tearDown method still needs to be implemented, since the POST
+request will save the image in the media folder.
+
+````python
+from django.core.files.uploadedfile import SimpleUploadedFile
+from io import BytesIO
+from PIL import Image
+import shutil
+
+class BookCreateViewTestCase(APITestCase):
+    def setUp(self):
+        self.data = {
+            "cover_image": #...
+            #...
+        }
+
+    def tearDown(self):
+        shutil.rmtree(REMOVE_MEDIA_FILES_PATH)
+
+    def test_create_with_image(self):
+        image = Image.new(mode="RGB", size=(20, 20))
+
+        tempFile = BytesIO()
+        image.save(tempFile, format="JPEG")
+        tempFile.seek(0)
+
+        with tempFile as temp:
+            image = SimpleUploadedFile("sample.jpg", temp.read(), content_type="image/jpeg")
```

Comment:
can we have a different variable name? either here or on line 130, it makes it more readable

Reply:
@pedro-inf-custodio ☝️

Reply:
something like `read_image` for the simpleuploadedfile one, and I would probably change the one on line 130 to `base_image` just to make sure there's no ambiguity between them

Reply:
updated